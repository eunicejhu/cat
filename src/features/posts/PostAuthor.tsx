import React from "react";
import { useSelector } from "react-redux";
import { State } from "../../store";

const PostAuthor: React.FC<{ id: string }> = ({ id }) => {
  const user = useSelector((state: State) =>
    state.users.find((user) => user.id === id)
  );
  return (
    <span className="post-author">
      By {user ? user.name : "Unknown author"}
    </span>
  );
};
export default PostAuthor;
