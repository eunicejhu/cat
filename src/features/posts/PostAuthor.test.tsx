import React from "react";
import PostAuthor from "./PostAuthor";
import renderer from "react-test-renderer";
import StoreWrapper from "../../test/StoreWrapper";
import { render } from "@testing-library/react";

test("render correctly", () => {
  const tree = renderer
    .create(
      <StoreWrapper>
        <PostAuthor id={"2"} />
      </StoreWrapper>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("show By Unknown author if user does not exist", () => {
  const { getByText } = render(
    <StoreWrapper>
      <PostAuthor id={"unknown"} />
    </StoreWrapper>
  );
  expect(getByText(/Unknown author/i)).toBeInTheDocument();
});
