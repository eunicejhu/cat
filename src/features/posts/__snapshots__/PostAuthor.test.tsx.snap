// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<span
  className="post-author"
>
  By 
  Madison Price
</span>
`;
